common/src/main/java/org/dash/wallet/common/services/TransactionMetadataProvider.kt

```diff
@@ -47,30 +47,55 @@ interface TransactionMetadataProvider {
     suspend fun getAllTransactionMetadata(): List<TransactionMetadata>
 
     // Address methods
+    /**
+     * mark an address with a tax category.  This will replace existing data
+     *
+     * @param address the address to mark
+     * @param isInput the address is an input in a transaction
+     * @param taxCategory the tax category
+     * @param service the name of the service associated with this address
+     */
```

Comment:
added this documentation to explain the API.

After maybe two months, I had forgotten some of the basics and it was unclear how things were working or not working.